tasks/prime.py

```diff
@@ -36,75 +36,42 @@ def fetch_mto_updates(self):
         except (json.JSONDecodeError, TypeError):
             logger.exception("Non-JSON response")
         else:
-            logger.info(f"ℹ️ Num MTOs returned: {len(json_body)}")
+            if str(resp.status_code).startswith("2"):
```

Comment:
I did this because I wanted it to be applicable and easy to copy over for all endpoints (we have some with that return response code 201 instead of 200 for a success - I don't know what this difference means, but alas!). But what I should really do if I want to make this reusable is create a response function like so:

```python
def check_response(task_name, response):
    logger.info(f"ℹ️  {task_name} status code: {response.status_code}")

    try:
        json_body = json.loads(response.content)
    except (json.JSONDecodeError, TypeError):
        logger.exception("Non-JSON response")
        return None

    if str(response.status_code).startswith("2"):
        logger.info(f"ℹ️  {task_name} successfully completed!")
    else:
        logger.error(f"⚠️ {json_body}")
        logger.error(payload)

    return json_body
```

Does that seem like a better solution to you? That's definitely the direction my heart wants to go in. But if you think we should keep it separate to allow for different error handling, that's valid too

Reply:
I pushed up the consolidated, function version of this! Let me know your thoughts, and I can change it back if you think that's more appropriate 😄

Reply:
looks awesome! great idea and implementation!